client/cmdhfmf.c

```diff
@@ -1100,8 +1100,12 @@ int CmdHF14AMfChk(const char *Cmd)
 	if (param_getchar(Cmd, 0)=='*') {
 		SectorsCnt = ParamCardSizeSectors(param_getchar(Cmd + 1, 0));
 	}
-	else
+    else {   
 		blockNo = param_get8(Cmd, 0);
+        // Singe Key check, so Set Sector count to cover sectors (1 to sector that contains the block)
+        SectorsCnt = (blockNo/4) + 1;   
+        SingleKey  = true;              // Set flag for single key check
+    }

```

Comment:
What if `Keytype == 2`? Is this correctly handled with `SingleKey = true`?

Reply:
This is interesting.  As I was not changing that code I never really thought about it.  Having a closer look the following line is where the check for a single block happens.
res = mfCheckKeys(blockNo, keyAB & 0x01, true, size, &keyBlock[6 * c], &key64);
So KeyAB & 1 would give  0 or a 1 and KeyAB is set to KeyValue, so a value of 2 would lead to the A key check.
I will add some code such that if keytype = 2 (?) then check both A and B

Reply:
Just tested the '?' and it does find both keys with the '?' (keytype 2)
So 'A' will just test for the 'A' key, 'B' for the b key and '?' for both.

Reply:
I have also renamed SingleKey to singleBlock (as we can check for 1 or 2 keys for a block check.